File: MaWPictures/src/main/java/us/mikeandwan/photos/di/AdapterModule.java
package us.mikeandwan.photos.di;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;
import us.mikeandwan.photos.services.DataServices;
import us.mikeandwan.photos.ui.categories.ICategoryListActivity;
import us.mikeandwan.photos.ui.categories.ListCategoryRecyclerAdapter;
import us.mikeandwan.photos.ui.categories.ThumbnailCategoryRecyclerAdapter;
import us.mikeandwan.photos.ui.photos.FullScreenImageAdapter;
import us.mikeandwan.photos.ui.photos.IPhotoActivity;
import us.mikeandwan.photos.ui.photos.ThumbnailRecyclerAdapter;


@Module
class AdapterModule {
    @Provides
    @PerActivity
    ListCategoryRecyclerAdapter provideListCategoryRecyclerAdapter(Activity activity,
                                                                   DataServices dataServices) {
        return new ListCategoryRecyclerAdapter((ICategoryListActivity) activity, dataServices);
    }


    @Provides
    @PerActivity
    ThumbnailCategoryRecyclerAdapter provideThumbnailCategoryRecyclerAdapter(Activity activity,
                                                                             DataServices dataServices) {
        return new ThumbnailCategoryRecyclerAdapter((ICategoryListActivity) activity, dataServices);
    }


    @Provides
    @PerActivity
    FullScreenImageAdapter provideFullScreenImageAdapter(Activity activity,
                                                         DataServices dataServices) {
        return new FullScreenImageAdapter((IPhotoActivity) activity, dataServices);
    }


    @Provides
    @PerActivity
    ThumbnailRecyclerAdapter provideThumbnailRecyclerAdapter(Activity activity,
                                                             DataServices dataServices) {
        return new ThumbnailRecyclerAdapter((IPhotoActivity) activity, dataServices);
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/di/AdapterModule.kt
package us.mikeandwan.photos.di

import android.app.Activity
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent
import dagger.hilt.android.scopes.ActivityScoped
import us.mikeandwan.photos.services.DataServices
import us.mikeandwan.photos.uiold.categories.ICategoryListActivity
import us.mikeandwan.photos.uiold.categories.ListCategoryRecyclerAdapter
import us.mikeandwan.photos.uiold.categories.ThumbnailCategoryRecyclerAdapter
import us.mikeandwan.photos.uiold.photos.FullScreenImageAdapter
import us.mikeandwan.photos.uiold.photos.IPhotoActivity
import us.mikeandwan.photos.uiold.photos.ThumbnailRecyclerAdapter
import us.mikeandwan.photos.uiold.receiver.ReceiverRecyclerAdapter

@Module
@InstallIn(ActivityComponent::class)
internal class AdapterModule {
    @Provides
    @ActivityScoped
    fun provideListCategoryRecyclerAdapter(
        activity: Activity,
        dataServices: DataServices
    ): ListCategoryRecyclerAdapter {
        return ListCategoryRecyclerAdapter(activity as ICategoryListActivity, dataServices)
    }

    @Provides
    @ActivityScoped
    fun provideThumbnailCategoryRecyclerAdapter(
        activity: Activity,
        dataServices: DataServices
    ): ThumbnailCategoryRecyclerAdapter {
        return ThumbnailCategoryRecyclerAdapter(activity as ICategoryListActivity, dataServices)
    }

    @Provides
    @ActivityScoped
    fun provideFullScreenImageAdapter(
        activity: Activity,
        dataServices: DataServices
    ): FullScreenImageAdapter {
        return FullScreenImageAdapter(activity as IPhotoActivity, dataServices)
    }

    @Provides
    @ActivityScoped
    fun provideThumbnailRecyclerAdapter(
        activity: Activity,
        dataServices: DataServices
    ): ThumbnailRecyclerAdapter {
        return ThumbnailRecyclerAdapter(activity as IPhotoActivity, dataServices)
    }

    @Provides
    @ActivityScoped
    fun provideReceiverRecyclerAdapter(activity: Activity): ReceiverRecyclerAdapter {
        return ReceiverRecyclerAdapter(activity)
    }
}